config/initializers/foreman.rb

```diff
@@ -1,26 +1,16 @@
 # stdlib dependencies
 require 'English'
 
-# Registries from app/registries/
-# All are loaded and populated early but are loaded only once
-require_dependency 'foreman/access_permissions'
-require_dependency 'foreman/plugin'
-require_dependency 'foreman/settings'
-
-# Other internal dependencies, may be autoloaded
-require 'net'
-require 'foreman/provision'
+# Registries from app/registries/ that do not create a namespace
+# should be loaded manually due to Zeitwerk
+require 'foreman/access_permissions'
+require 'foreman/settings'
 
 Rails.application.config.before_initialize do
   # load topbar
   Menu::Loader.load
 end
 
-Foreman.settings.load_definitions
```

Comment:
Would this part address https://projects.theforeman.org/issues/33888?

Reply:
I think so, it seems this gets executed as part of https://github.com/theforeman/foreman/blob/develop/config/initializers/foreman.rb#L37 anyway. It also contains `Foreman.settings.load_values`, which is called in `after_initialize` callback. I think that is redundant...

I mean, I'm not 100% sure why we have it as we do, but for me it seems that `Foreman.settings.load` in `to_prepare` is sufficient and does the same thing as combination of `Foreman.settings.load_definitions` here and `Foreman.settings.load_values` in `after_initialize`. For me it also seems like currently we try to load the definitions and values 2 times.

Should I change the commit message for this?

Reply:
Like https://github.com/theforeman/foreman/pull/10131#discussion_r1596743330.

Reply:
I'll leave it for now here as is, 'cause I didn't yet tested these new changes with plugins (working on it though).